test/infrastructure/docker/internal/loadbalancer/config.go

```diff
@@ -55,23 +57,30 @@ defaults
   default-server init-addr none
 
 frontend control-plane
-  bind *:{{ .ControlPlanePort }}
+  bind *:{{ .FrontendControlPlanePort }}
   {{ if .IPv6 -}}
-  bind :::{{ .ControlPlanePort }};
+  bind :::{{ .FrontendControlPlanePort }};
   {{- end }}
   default_backend kube-apiservers
 
 backend kube-apiservers
   option httpchk GET /healthz
   # TODO: we should be verifying (!)
   {{range $server, $address := .BackendServers}}
-  server {{ $server }} {{ $address }} check check-ssl verify none resolvers docker resolve-prefer {{ if $.IPv6 -}} ipv6 {{- else -}} ipv4 {{- end }}
+  server {{ $server }} {{ JoinHostPort $address $.BackendControlPlanePort }} check check-ssl verify none resolvers docker resolve-prefer {{ if $.IPv6 -}} ipv6 {{- else -}} ipv4 {{- end }}
   {{- end}}
 `
 
 // Config generates the loadbalancer config from the ConfigTemplate and ConfigData.
-func Config(data *ConfigData) (config string, err error) {
-	t, err := template.New("loadbalancer-config").Parse(ConfigTemplate)
+func Config(data *ConfigData, customTemplate string) (config string, err error) {
+	configTemplate := DefaultTemplate
+	if customTemplate != "" {
+		configTemplate = customTemplate
+	}
+
+	t, err := template.New("loadbalancer-config").Funcs(template.FuncMap{
+		"JoinHostPort": net.JoinHostPort,
+	}).Parse(configTemplate)
```

Comment:
Wondering if we should keep it simple in here and passing the `DefaultTemplate` already from the above func.

```suggestion
func Config(data *ConfigData, configTemplate string) (config string, err error) {
	t, err := template.New("loadbalancer-config").Funcs(template.FuncMap{
		"JoinHostPort": net.JoinHostPort,
	}).Parse(configTemplate)
```